presto-mysql/src/test/java/io/prestosql/plugin/mysql/TestMySqlTypeMapping.java

```diff
@@ -219,6 +239,101 @@ public void testDecimalExceedingPrecisionMax()
         testUnsupportedDataType("decimal(50,0)");
     }
 
+    @DataProvider
+    public Object[][] testDecimalExceedingPrecisionMaxProvider()
+    {
+        return new Object[][] {
+                {60, 20},
+                {65, 25},
+        };
+    }
+
+    @Test(dataProvider = "testDecimalExceedingPrecisionMaxProvider")
+    public void testDecimalExceedingPrecisionMaxWithDecimalMapping(Integer typePrecision, Integer typeScale)
+    {
+        JdbcSqlExecutor jdbcSqlExecutor = new JdbcSqlExecutor(mysqlServer.getJdbcUrl());
+
+        List<String> testData = asList(
+                "'exact', 1.12345678",
+                "'exact',-1.12345678",
+                "'round', 12345678901234567890.12345678901234567890",
+                "'round',-12345678901234567890.12345678901234567890",
+                "'unsupported', 1234567890123456789012345678901234567890.1234567890",
+                "'unsupported',-1234567890123456789012345678901234567890.1234567890");
+
+        try (TestTable testTable = new TestTable(
+                jdbcSqlExecutor,
+                "tpch.test_exceeding_max_decimal",
+                format("(label varchar(12), d_col decimal(%d,%d))", typePrecision, typeScale),
+                testData)) {
+            String selectSchema = format("SELECT column_name, data_type FROM information_schema.columns WHERE table_schema = 'tpch' AND table_name = '%s'", testTable.getName().split("[.]")[1]);
+            String selectSupportedRows = format("SELECT d_col FROM %s WHERE label <> 'unsupported'", testTable.getName());
+            Function<String, String> selectRows = (label) -> format("SELECT d_col FROM %s WHERE label like '%s'", testTable.getName(), label);
+            Function<Integer, String> expectedSchema = (scale) -> format("VALUES ('label','varchar(12)'), ('d_col', 'decimal(%d,%d)')", Decimals.MAX_PRECISION, scale);
+            Function<String, String> expectedValues = (label) -> "VALUES " + testData.stream()
+                    .filter(s -> s.startsWith(format("'%s',", label)))
+                    .map(s -> s.split(",")[1])
+                    .collect(joining(","));
+            Function<List<Object>, String> expectedRoundedValues = (rounding) -> "VALUES " + testData.stream()
+                    .filter(s -> !s.startsWith("'unsupported',"))
+                    .map(s -> new BigDecimal(s.split(",")[1].trim()).setScale((Integer) rounding.get(0), (RoundingMode) rounding.get(1)).toString())
+                    .collect(joining(","));
+
+            int mappingScale;
+            for (int scale : asList(0, 7)) {
+                mappingScale = calcDecimalScaleMapping(typePrecision, typeScale, scale);
```

Comment:
I don't like that you verify product code with product code. I would much prefer if you would hard coded expected values here.

It should be clear what is the result precision and scale when for example mysql has `decimal(60,10)` and decimal default scale is set to 7. It is like documenting the behaviour. Now test expectations change whenever the `calcDecimalScaleMapping` changes. We could regress not knowing about it.